src/kibana/plugins/vis_types/vislib/_vislib_renderbot.js

```diff
@@ -7,13 +7,14 @@ define(function (require) {
     _(VislibRenderbot).inherits(Renderbot);
     function VislibRenderbot(vis, $el) {
       VislibRenderbot.Super.call(this, vis, $el);
-      this.vislibVis = {};
       this._createVis();
     }
 
     VislibRenderbot.prototype._createVis = function () {
       var self = this;
-
+      if (self.vislibVis) {
+        self.destroy();
```

Comment:
@rashidkpc this _vislib_renderbot.js fix was submitted in pr #3587.
@spalger and @w33ble helped track this down. It was likely a factor in some memory leak issues, and it was causing a fatal error in tile map after switching from heatmap to another map type and then resizing browser window. 
Previous chart (map) was not being destroyed properly by renderbot.